service-base/src/main/java/org/eclipse/hono/service/metric/MetricsTags.java

```diff
@@ -241,6 +243,82 @@ public Tag asTag() {
         }
     }
 
+    /**
+     * A status providing more information about the outcome of processing a message received from a device.
+     * <p>
+     * The immutable enum check is disabled because the offending field's (tag) type
+     * is in fact immutable but has no corresponding annotation.
+     */
+    @SuppressWarnings("ImmutableEnumChecker")
+    public enum Reason {
+
+        /**
+         * The message received has a bad syntax and could not be parsed.
+         */
+        BAD_SYNTAX("bad-syntax"),
+        /**
+         * The type of the message could not be deducted from the message.
+         */
+        UNKNOWN_TYPE("unknown-type"),
+        /**
+         * The type of the message is not supported for this action.
+         */
+        UNSUPPORTED_TYPE("unsupported-type"),
```

Comment:
no, the first 3 do not.

Reply:
I believe the others are useful as well. It provides a lot more insight into missed messages. Lot's of our devices are out in the field an unreachable. Having some reporting from the cloud services makes sense imo.

Reply:
The only concern I have is that these kinds of issues (currently) only occur with the Lora adapter because none of the other adapters actually care about the message payload. In fact, being payload agnostic is actually one of Hono's core features/assertions ...

Reply:
Agreed, but it is very useful information. If a message is unprocessable because it was wrongly formatted, it's completely different from a unknown type. From an operational standpoint, we need to act differently

Reply:
It doesn't hurt, right? It only adds information in the metrics. You are free to use it or not.